pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -186,21 +186,40 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 			})
 			return nil
 		}
-		return fmt.Errorf("error getting Tasks and/or TaskRuns for Pipeline %s: %s", p.Name, err)
+		return fmt.Errorf("error getting Tasks for Pipeline %s: %s", p.Name, err)
 	}
-	prtr := resources.GetNextTask(pr.Name, pipelineState, c.Logger)
+	for _, rprt := range pipelineState {
+		err := taskrun.ValidateResolvedTaskResources(rprt.PipelineTask.Params, rprt.ResolvedTaskResources)
+		if err != nil {
+			c.Logger.Error("Failed to validate pipelinerun %s with error %v", pr.Name, err)
+			pr.Status.SetCondition(&duckv1alpha1.Condition{
+				Type:    duckv1alpha1.ConditionSucceeded,
+				Status:  corev1.ConditionFalse,
+				Reason:  ReasonFailedValidation,
+				Message: err.Error(),
+			})
+			return nil
+		}
+	}
+	err = resources.ResolveTaskRuns(c.taskRunLister.TaskRuns(pr.Namespace).Get, pipelineState)
```

Comment:
I feel this func is not really doing much, but just here for the test!

Reply:
I felt kind of weird about it too! This logic used to be inside of `ResolvePipelineRun` - I think it makes sense to break it out but I'm not sure happy with how it looks right now :S Do you have any other ideas?

I'm thinking we might want to create a type that hold the `PipelineState` and have functions like this operate on it maybe :J that's the only idea I have right now

Reply:
Having some internal types for ResolvedTasks would be useful

Reply:
we can iterate on that one, but lets merge it for now